test/e2e/gardener/shoot/create_migrate_delete.go

```diff
@@ -70,9 +72,11 @@ var _ = Describe("Shoot Tests", Label("Shoot", "control-plane-migration"), func(
 		ItShouldInitializeSeedClient(s)
 
 		It("Verify that all secrets have been migrated without regeneration", func(ctx SpecContext) {
-			secretsAfterMigration, err := GetPersistedSecrets(ctx, s.SeedClientSet, s.Shoot.Status.TechnicalID)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(ComparePersistedSecrets(secretsBeforeMigration, secretsAfterMigration)).To(Succeed())
+			Eventually(ctx, func(g Gomega) {
+				secretsAfterMigration, err := GetPersistedSecrets(ctx, s.SeedClientSet.Client(), s.Shoot.Status.TechnicalID)
+				g.Expect(err).ToNot(HaveOccurred())
+				g.Expect(ComparePersistedSecrets(secretsBeforeMigration, secretsAfterMigration)).To(Succeed())
```

Comment:
I'm wondering whether this check should be after the `Eventually`. I.e., we could only retry the secret retrieval until it succeeds and then we do the comparison.